api.go

```diff
@@ -134,7 +135,32 @@ type ExecutionPayload struct {
 	Transactions []Data `json:"transactions"`
 }
 
-type PayloadAttributes struct {
+func (p *ExecutionPayloadV1) ValidateRoot() bool {
+	header := &types.Header{
+		ParentHash:  p.ParentHash,
+		UncleHash:   common.Hash{},
+		Coinbase:    p.FeeRecipient,
+		Root:        common.Hash(p.StateRoot),
+		TxHash:      common.Hash{},
```

Comment:
This hash should be set: `types.EmptyRootHash` for an empty tx list. But if it will be non-empty, then: `types.DeriveSha(types.Transactions(txs), hasher)`